lib/crono_trigger/polling_thread.rb

```diff
@@ -55,12 +52,15 @@ def alive?
     def poll(model)
       @logger.info "(polling-thread-#{Thread.current.object_id}) Poll #{model}"
 
+      queue_empty_event = Concurrent::Event.new
       maybe_has_next = true
       while maybe_has_next && !@stop_flag.set?
+        queue_empty_event.wait unless @executor.queue_length.zero?
```

Comment:
[nits]

It may wait indefinitely if these statements are processed in the following order.

1. `@executor.queue_length.zero?` return false
1. All executors finished at the same time.
1. evaluate `queue_empty_event.wait`

But, I think that it is unlikely to occur.
There is no problem in practice.

Reply:
It never happens because, once `queue_empty_event.set`  is executed, `queue_empty_event.wait` doesn't wait.
The fact that `@executor.queue_length.zero?` returns false means that `queue_empty_event.set` will certainly be executed at least once.

Reply:
Thanks.
I've understood.